### ISSUE TITLE
Column header next to 'MW-score for each function' is missing? 

### ISSUE THREAD

**Issue by Anto007** [AnantharamanLab/METABOLIC#69]
Fantastic tool and thank you so much for bringing out this to the community. I'm attaching here the MW-score result file and the output log files that I got after running the METABOLIC-C module. Strangely, the third column header is missing from the MW_score_result.txt file and I'm left scratching my head to understand what the values in the third column (without the header) actually represent? Any inputs here would be very much appreciated
[MW-score_result.txt](https://github.com/AnantharamanLab/METABOLIC/files/8488748/MW-score_result.txt)
[METABOLIC_log.log](https://github.com/AnantharamanLab/METABOLIC/files/8488749/METABOLIC_log.log)
[METABOLIC_run.log](https://github.com/AnantharamanLab/METABOLIC/files/8488751/METABOLIC_run.log)

.

**Comment by Anto007** [comment]
Also, I find this empty first taxon issue replicated in the other figures from running METABOLIC-C.pl on my datasets. Please find attached the same here; thanks again!
[Metabolic_Sankey_diagram.pdf](https://github.com/AnantharamanLab/METABOLIC/files/8488794/Metabolic_Sankey_diagram.pdf)
[Functional_network_figure.pdf](https://github.com/AnantharamanLab/METABOLIC/files/8488796/Functional_network_figure.pdf)

**Comment by patriciatran** [comment]
Hi @Anto007 , 

Thank you for trying this tool! You can read more about MW-table here: which you can see an example of here on Slide 7: https://github.com/AnantharamanLab/METABOLIC/blob/master/METABOLIC_C-DEBI_slides.pdf 

In your case, the third column is actually probably the taxa that is unassigned to the phylum level -- which is related to your second question:

As for the empty taxa (both in the MW-score table and the Figures issue), can you check in the Intermediate file what you have for the GTDBTK classification? IF there was a "find and replace" to remove "p__" in front of the taxa names, and GTDBTk could only assign it to no phylum, this might be why the MAGS are shown as no phylum in the Figures.

Let me know if that helps!

Patricia

**Comment by Anto007** [comment]
Thanks a zillion for your super-fast response. If I understood what you meant correctly, I don't think that I have any MAGs where the GTDBTK classification did not output some text after "p__". All of the MAGs do appear to have some phylum name assigned and I'm attaching here the relevant intermediate files (modified to .txt format to comply with uploading here). Please feel free to let me know in case you need me to share any of the other intermediate files that were generated during my run. Thanks again
[gtdbtk.ar122.summary.tsv.txt](https://github.com/AnantharamanLab/METABOLIC/files/8489699/gtdbtk.ar122.summary.tsv.txt)
[gtdbtk.bac120.summary.tsv.txt](https://github.com/AnantharamanLab/METABOLIC/files/8489701/gtdbtk.bac120.summary.tsv.txt)

**Comment by ChaoLab** [comment]
Hi @Anto007,
I think it is like a format thing for your case. If GTDB-Tk does assign all the MAGs with phylum names (no matter a real name or an alphanumeric one), there will be no error reports as shown by you. Are 86 archaeal MAGs and 243 bacterial MAGs the whole dataset? Maybe you can do some troubleshootings with a subset of them, to begin with

**Comment by Anto007** [comment]
Thanks for your kind response @patriciatran and @ChaoLab. A total of 334 MAGs/fasta files were input by me to the METABOLIC pipeline. Out of this, the early lines in the gtdbtk log report 248 to be bacteria and 86 to be archaea
[gtdbtk.log](https://github.com/AnantharamanLab/METABOLIC/files/8515862/gtdbtk.log)

However, the log towards the end says the below. Does this mean 5 bacterial MAGs (248-243) were left out? However, the confusing thing still is that all bacterial MAGs seems to have classification until the phylum-level in [gtdbtk.bac120.summary.tsv.txt](https://github.com/AnantharamanLab/METABOLIC/files/8489701/gtdbtk.bac120.summary.tsv.txt)
```
[2022-04-14 00:41:32] TASK: Placing 248 bacterial genomes into reference tree with pplacer using 34 CPUs (be patient).
[2022-04-14 00:41:33] INFO: pplacer version: v1.1.alpha19-0-g807f6f3
[2022-04-14 01:24:15] INFO: Calculating RED values based on reference tree.
[2022-04-14 01:24:25] TASK: Traversing tree to determine classification method.
[2022-04-14 01:24:27] INFO: Completed 243 genomes in 1.26 seconds (193.28 genomes/second).
```

**Comment by Anto007** [comment]
Oops I just realized that there are only 243 MAGs in [gtdbtk.bac120.summary.tsv.txt](https://github.com/AnantharamanLab/METABOLIC/files/8489701/gtdbtk.bac120.summary.tsv.txt) and so this means that the third column was indeed representing MAGs that did not have a phylum-level assignment. Apologies for not spotting this earlier. Closing this issue and thanks a zillion again for your wonderful support-much appreciated!
